src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -1164,7 +1172,26 @@ protected void writeItem(Object item, MarshallingContext context,
                 }
                 context.convertAnother( item, new ReferenceConverter( clazz ) );
             } else if (writer instanceof JettisonStaxWriter) {
+                /*
+                 * GEOS-7771 / GEOS-7873
+                 * Workaround for an array serialization bug in jettison 1.0.1
+                 * Can be removed when we upgrade to jettison 1.2
+                 */
                 writer.setValue("null");
+                try {
+                    Field outField = StaxWriter.class.getDeclaredField("out");
+                    Field nodesField = MappedXMLStreamWriter.class.getDeclaredField("nodes");
+                    outField.setAccessible(true);
+                    nodesField.setAccessible(true);
+                    FastStack nodes = (FastStack) nodesField.get(outField.get(writer.underlyingWriter()));
+                    if (nodes.peek() instanceof JSONArray) {
+                        nodes.pop();
+                    }
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
```

Comment:
Ouch, did we really merge something writing stack traces to the standard error instead of using logging? Checked the code, indeed we did :-(